src/lane_keep_assist/scripts/lane_turn.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+import cv2 as cv
+import numpy as np
+import time
+import math
+import lane_detect
+from typing import List, Tuple
+from sensor_msgs.msg import Image
+from cv_bridge import CvBridge, CvBridgeError
+
+
+# Returns the results of both detection methods lane and approximate road boundaries
+# Returns the percent difference between the two methods detection if they agree on what type of detection
+# Returns the coordinates of the two lines
+# Returns the current majority segment found in the segmentation image by the sum of the RGB value
+def lane_interpret(img: np.ndarray, segmented_img: np.ndarray):
+    bridge = CvBridge()
+    height = img.height
+
+    if img and segmented_img:
+        try:
+            cv_img = bridge.imgmsg_to_cv2(img, desired_encoding="passthrough")
+            cv_segmented_img = bridge.imgmsg_to_cv2(segmented_img, desired_encoding="passthrough")
+
+            # Get a result from both the normal camera image and the segmentation image
+            gradient_lanes, hls_lanes, segmentation_lanes, road_colour = lane_detect.lane_detect(cv_img, cv_segmented_img)
+
+            # Get the lane results
+            gradient_result = lane_type(gradient_lanes, height)
+            hls_result = lane_type(hls_lanes, height)
+            segmentation_result = lane_type(segmentation_lanes, height)
+
+            # Get the percent differences for the detections
+            gradient_difference = compare_lanes(gradient_lanes, segmentation_lanes, gradient_result,
+                                                segmentation_result)
+            hls_difference = compare_lanes(hls_lanes, segmentation_lanes, hls_result, segmentation_result)
+
+            return ((gradient_result, hls_result, segmentation_result), (gradient_difference, hls_difference),
+                    (format_lanes(gradient_lanes), format_lanes(hls_lanes), format_lanes(segmentation_lanes)), road_colour)
+
+        except CvBridgeError as e:
+            return (4, 4, 4), (100, 100), ([], [], []), 1
+
+
+# Returns the type of lane detection
+# 1 is two lanes
+# 2 is one lane off to the left
+# 3 is one lane off to the right
+# 4 is no lanes
+def lane_type(lines, height):
+    if lines:
+        if len(lines) == 2:
```

Comment:
Why not use an Enum?

Reply:
good point, meant to do this by forgot, will add.